client/app/services/group/currentGroup.spec.js

```diff
@@ -14,36 +14,61 @@ describe("CurrentGroup service", () => {
     $log.assertEmpty();
   });
 
-  let CurrentGroup;
+  let CurrentGroup, Authentication, $httpBackend, stub;
 
   beforeEach(inject(($injector) => {
     CurrentGroup = $injector.get("CurrentGroup");
+    Authentication = $injector.get("Authentication");
+    $httpBackend = $injector.get("$httpBackend");
+
+    Authentication.data = { id: 1 };
+    stub = sinon.stub(CurrentGroup, "persistCurrentGroup");
   }));
 
+  afterEach(() => {
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
   it("starts with an empty empty", () => {
     expect(CurrentGroup.value).to.deep.equal({});
   });
 
   it("can be set", () => {
     let newGroup = {
+      id: 1,
       anything: "can",
       go: "here",
       including: ["lists", "of", "stuff"]
     };
     CurrentGroup.set(newGroup);
+    assert(stub.withArgs(newGroup.id).calledOnce);
```

Comment:
Style thing: we only have `expect` in our codebase, maybe should stick to it for consistence? [using chai](https://github.com/domenic/sinon-chai)
`expect(CurrentGroup.persistCurrentGroup).to.have.been.calledOnce`

Also, I find it better to not introduce new references for the stub: `CurrentGroup.persistCurrentGroup` instead of `stub`